### PR TITLE
CO-2358: [Spike]: How to do E2E test

### PR DESCRIPTION
**Pull request by damianimashaun**
## Description
[Spike]: How to do E2E test
Using detox + GitHub actions
Created a Small MVP app
Added CI workflows 

## Jira issue
[CO-2358](https://islamicfinanceguru.atlassian.net/browse/CO-2358)

[CO-2358]: https://islamicfinanceguru.atlassian.net/browse/CO-2358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/e2e-ios.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: macOS-latest
+    timeout-minutes: 60
+    
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14"
```
Comment (shuaibiyy) [inline .github/workflows/e2e-ios.yml:18]: does this work with node 18?

File: .github/workflows/e2e-ios.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Run Detox Build
+        run: yarn e2e:build-ios
+
+      - name: Install Detox Dependencies
+        run: |
+          brew tap wix/brew
+          brew install applesimutils
+
+      - name: Run Detox Tests
```
Comment (mahmoud16h) [inline .github/workflows/e2e-ios.yml:37]: is it possible to cobine both android and ios into one workflow? reason being is we're doing quite a few of the same things twice, such as `yarn install` 'unit tests' and detox dependencies (although the detox dependencies don't seem to be installed in android.

File: .github/workflows/e2e-ios.yml (hunk 1 above)
Comment (damianimashaun) [reply .github/workflows/e2e-ios.yml]: yes, fixed it.

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+name: End to end tests
```
Comment (mahmoud16h) [inline .github/workflows/e2e.yml:1]: I think we can just rename this to `Tests` as it runs the unit tests as well as e2e... but great stuff that's about it 😀

**Pull request merged**